### PR TITLE
[#465] Add Arkana to store keys and secrets away from source code

### LINKED ISSUE #465: Add Arkana to store keys and secrets away from source code
## Why

Our iOS template doesn't have a way to store keys and secrets. In some projects, we used [Cocoapods-keys](https://github.com/orta/cocoapods-keys), which has many disadvantages compared to [Arkana](https://github.com/rogerluan/arkana). With Arkana, we can integrate it into our CI/CD very easily with .env files.
 
## Who Benefits?

- Developers

### PR DESCRIPTION
**Pull request by ducbm051291**
- Close #465 

## What happened 👀

Integrate Arkana to the ios template with the followings:
- Add `.arkana.yml` (Arkana Configuration file)
- Add `.env.example` (Example of .env file for Arakana)
- Add Arkana to `Gemfile` for installing bundles
- Update `make.sh` to run Arkana after installing bundles
- Update Podfile with 2 Generated Pods by Arkana
- Add sample usage in Constant+API.swift how to use Arkana
- Update workflows to run arkana

## Insight 📝

How to test: 

1. Create a `.env` file from the `.env.example` file
2. Run `make.sh`

## Proof Of Work 📹

**Run make.sh successfully with Arkana**
CleanShot 2023-04-07 at 10 35 56

**Build project successfully with Arkana**
CleanShot 2023-04-07 at 10 37 48

### CODE REVIEW

File: .arkana.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+result_path: 'dependencies' # Optional. Destination path of the generated code, related to the path of this config file. Defaults to ArkanaKeys.
```
Comment (blyscuit) [inline .arkana.yml:1]: I suggestion a capital naming for this folder `Dependencies`.
I also think the default naming makes sense as is 🤔 
``` podfile
  pod "ArkanaKeys", path: "./.ArkanaKeys/ArkanaKeys"
```

File: .arkana.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+swift_declaration_strategy: let # Optional. One of: lazy var, var, let. Defaults to let.
+should_generate_unit_tests: true # Optional. One of: true, false. Defaults to true.
+package_manager: cocoapods # Optional. One of: spm, cocoapods. If you use both, declare cocoapods. Defaults to spm.
+environments: # Optional. List of environments that will be used to generate secret keys when you have keys that are different between environments (e.g. debug/staging/prod). Defaults to empty.
+  - Staging
+  - Release
+global_secrets:
+environment_secrets:
+  - ClientId
+  - ClientSecret
```
Comment (blyscuit) [inline .arkana.yml:12]: I think we should make them minimal, documentation can be seen on the Arkana's repo. Keeping documentation here could be deprecated if Arkana changes their. 🙏 
```suggestion
package_manager: cocoapods
environments:
  - Staging
  - Release
global_secrets:
environment_secrets:
```

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,4 @@
+ClientIdStaging = "SET YOUR VALUE HERE"
```
Comment (blyscuit) [inline .env.example:1]: I think we can remove these values to make it more minimal.

File: {PROJECT_NAME}/Sources/Constants/Constants+API.swift [hunk 4]
```diff
@@ -3,4 +3,15 @@
 //  Constants+API.swift
 //
 
-extension Constants.API {}
+import ArkanaKeys
+
+extension Constants.API {
+
+    #if STAGING
```
Comment (blyscuit) [inline {PROJECT_NAME}/Sources/Constants/Constants+API.swift:10]: Same here, can revert file 🤔 .

**Review by blyscuit — COMMENTED** [review]
Some comment regarding making the template minimal. 🙏

**Review by blyscuit — APPROVED** [review]
Looks good.

**Review by vnntsu — APPROVED** [review]

**Pull request merged**